File: web/src/constants/syncStatus.ts
import type { SyncGenerationStatus, SyncStatus, SyncTrainingStatus } from '@/types'

export const ACTIVE_SYNC_STATUSES: readonly SyncStatus[] = [
  'syncing',
  'generating',
  'training',
  'loading_adapter',
]

export const SYNC_GENERATION_TOGGLE_STATUS: SyncGenerationStatus = 'completed'

export const SYNC_TRAINING_SUCCEEDED_STATUSES: readonly SyncTrainingStatus[] = [
  'completed',
  'adapter_loaded',
  'adapter_unloaded',
  'adapter_load_failed',
  'adapter_failed',
]

export const SYNC_TRAINING_CAN_LOAD_STATUSES: readonly SyncTrainingStatus[] = [
  'adapter_load_failed',
  'adapter_failed',
  'adapter_unloaded',
  'completed',
]

File: train_factory/enums/sync_status.py
"""External sync related statuses."""


class SyncStatus:
    """Sync task status."""

    IDLE = "idle"
    SYNCING = "syncing"
    GENERATING = "generating"
    TRAINING = "training"
    LOADING_ADAPTER = "loading_adapter"
    ERROR = "error"
    DELETING = "deleting"
    DELETING_CASCADE = "deleting_cascade"


class BatchStatus:
    """Sync batch status."""

    FETCHED = "fetched"
    REGISTERED = "registered"
    GENERATION_QUEUED = "generation_queued"
    GENERATION_DONE = "generation_done"


class SyncGenerationStatus:
    """Sync generation tracking status."""

    PENDING = "pending"
    COMPLETED = "completed"
    FAILED = "failed"


class SyncTrainingStatus:
    """Sync training tracking status."""

    PENDING = "pending"
    COMPLETED = "completed"
    FAILED = "failed"
    ADAPTER_LOADED = "adapter_loaded"
    ADAPTER_UNLOADED = "adapter_unloaded"
    ADAPTER_LOAD_FAILED = "adapter_load_failed"
    ADAPTER_FAILED = "adapter_failed"


class TrainingTargetStatus:
    """Training target status within a sync task."""
    IDLE = "idle"
    READY = "ready"
    TRAINING = "training"
    LOADING_ADAPTER = "loading_adapter"
    ERROR = "error"

